### PR TITLE
#2947

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/2947 - [studio-ui] Add Create button in the context nav #2947

### CODE REVIEW

File: static-assets/components/cstudio-admin/mods/content-types.js [hunk 1]
```diff
@@ -1561,6 +1561,10 @@ CStudioAuthoring.Module.requireModule(
 					});
 				}
 
+                this.createRowHeading(CMgs.format(langBundle, "quickCreate"), sheetEl);
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-types.js:1573]: Break long lines

File: static-assets/components/cstudio-common/common-api.js [hunk 2]
```diff
@@ -4493,6 +4501,26 @@ var nodeOpen = false,
...
             },
 
+            /**
+             * get Quick Create
+             */
+            getQuickCreate: function(callback) {
+                var serviceUrl = this.getQuickCreateURL;
+                serviceUrl += "?siteId=" + CStudioAuthoringContext.site;
+
+                var serviceCallback = {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Use new react app `ajax` lib. Like so:
```js
CrafterCMSNext.util.ajax.get(this.createServiceUri(serviceUrl))
  .subscribe(
    function (response) {
      var results = response.response;
      results = results.items;
      callback.success(results);
    },
    function (response) {
      callback.failure(response);
    }
  );
```

File: static-assets/components/cstudio-contextual-nav/quick-create.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+CStudioAuthoring.ContextualNav.WcmQuickCreate = CStudioAuthoring.ContextualNav.WcmQuickCreate || {
+
+    initialized: false,
+
+    /**
+     * initialize module
+     */
+    initialize: function(config) {
+        YAHOO.util.Event.onAvailable("quick-create", function() {
+            CStudioAuthoring.Service.getQuickCreate( {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/quick-create.js]: 1. Will it really ever not be available?
2. Don't wait for the element to be available to retrieve the data. Retrieve the data and initialize the element when it is available.

File: static-assets/components/cstudio-contextual-nav/quick-create.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            }else{
+                                CStudioAuthoring.Operations.refreshPreview();
+                            }
+                        },
+                        failure: function() { },
+                        callingWindow: window
+                    };
+                    if (response.length > 0 && (CStudioAuthoringContext.isPreview || CStudioAuthoringContext.isDashboard)) {
+                        $(quickCreateWrapper).removeClass('hide');
+                        for (var i = 0; i < response.length; i++) {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/quick-create.js:67]: Create the entire HTML chunk and then append it to the container.

File: static-assets/components/cstudio-contextual-nav/quick-create.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                                                createCb,
+                                                null
+                                            );
+                                        })
+                                    ));
+                            })(response[i]);
+                        }
+                    }
+                },
+                failure: function() {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/quick-create.js]: Handle error

File: templates/web/site-config.ftl [hunk 6]
```diff
@@ -58,7 +58,7 @@
 
 </head>
 
-<body class="yui-skin-cstudioTheme">
+<body class="yui-skin-cstudioTheme siteConfiguration">
```
Comment (rart) [inline templates/web/site-config.ftl]: What's this for?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-common/common-api.js (hunk 2 above)
Comment (rart) [reply static-assets/components/cstudio-common/common-api.js]: PR is now merged: https://github.com/craftercms/studio-ui/pull/1193

File: static-assets/components/cstudio-admin/mods/content-types.js (hunk 1 above)
Comment (vestrada) [reply static-assets/components/cstudio-admin/mods/content-types.js:1573]: Done

File: static-assets/components/cstudio-common/common-api.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: Done

File: static-assets/components/cstudio-contextual-nav/quick-create.js (hunk 3 above)
Comment (vestrada) [reply static-assets/components/cstudio-contextual-nav/quick-create.js]: Done

File: static-assets/components/cstudio-contextual-nav/quick-create.js (hunk 4 above)
Comment (vestrada) [reply static-assets/components/cstudio-contextual-nav/quick-create.js:67]: Done

File: static-assets/components/cstudio-contextual-nav/quick-create.js (hunk 5 above)
Comment (vestrada) [reply static-assets/components/cstudio-contextual-nav/quick-create.js]: Done

File: templates/web/site-config.ftl (hunk 6 above)
Comment (vestrada) [reply templates/web/site-config.ftl]: Done

File: static-assets/components/cstudio-common/common-api.js [hunk 7]
```diff
@@ -4493,6 +4501,27 @@ var nodeOpen = false,
...
+                CrafterCMSNext.util.ajax.get(this.createServiceUri(serviceUrl))
+                    .subscribe(
+                    function (response) {
+                        var results = response.response;
+                        results = results.items;
+                        callback.success(results);
+                    },
+                    function (response) {
+                        callback.failure(response);
+                        return null;
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Remove return statement.

File: static-assets/components/cstudio-admin/mods/content-types.js (hunk 1 above)
Comment (rart) [reply static-assets/components/cstudio-admin/mods/content-types.js:1573]: Please format code appropriately for readability.

File: static-assets/components/cstudio-contextual-nav/quick-create.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        var dropdown = $('#quick-create'),
+            quickCreateWrapper = $('.dropdown.quick-create'),
+
+            self = this;
+        CStudioAuthoring.Service.getQuickCreate({
+            success: function (response) {
+                var createCb = {
+                    success: function (contentTO, editorId, name, value, draft) {
+                        var page = CStudioAuthoring.Utils.getQueryParameterURL("page");
+                        var acnDraftContent = YDom.getElementsByClassName("acnDraftContent", null, parent.document)[0];
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/quick-create.js]: use `jQuery` instead

File: templates/web/site-config.ftl [hunk 9]
```diff
@@ -47,6 +47,7 @@
     <script src="/studio/static-assets/scripts/animator.js?version=${UIBuildId!.now?string('Mddyyyy')}"></script>
     <script src="/studio/static-assets/scripts/communicator.js?version=${UIBuildId!.now?string('Mddyyyy')}" ></script>
     <script src="/studio/static-assets/libs/js-cache/cache.js?version=${UIBuildId!.now?string('Mddyyyy')}"></script>
+    <script src="/studio/static-assets/libs/bootstrap/js/bootstrap.min.js?version=${UIBuildId!.now?string('Mddyyyy')}"></script>
```
Comment (rart) [inline templates/web/site-config.ftl:50]: jQuery ui tooltips are already in the runtime. Use that instead?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-common/common-api.js (hunk 7 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: Done

File: static-assets/components/cstudio-contextual-nav/quick-create.js (hunk 8 above)
Comment (vestrada) [reply static-assets/components/cstudio-contextual-nav/quick-create.js]: Done

File: templates/web/site-config.ftl (hunk 9 above)
Comment (vestrada) [reply templates/web/site-config.ftl:50]: I'm using popover not tooltips. Looks better.

**Review by rart — CHANGES_REQUESTED** [review]
Please remove built `css` assets which are now built automatically (🥳)

File: templates/web/site-config.ftl (hunk 9 above)
Comment (rart) [reply templates/web/site-config.ftl:50]: We're importing all of bootstrap to use 1 component. Let's import the component(s) we will use.

**Comment by vestrada** [comment]
They were already removed. Actually they were giving conflicts, so I removed them.

File: templates/web/site-config.ftl (hunk 9 above)
Comment (vestrada) [reply templates/web/site-config.ftl:50]: Per our conversation this will be address on other ticket.

Comment (vestrada) [reply templates/web/site-config.ftl]: Per our conversation this will be address on other ticket.

**Review by rart — APPROVED** [review]

**Pull request merged**